addons/website/models/website.py

```diff
@@ -437,37 +438,53 @@ def page_search_key_dependencies(self, page_id=False):
     # Languages
     # ----------------------------------------------------------
 
-    def get_alternate_languages(self, req=None):
+    def _get_alternate_languages(self, canonical_params):
+        self.ensure_one()
+
+        # Compare query string dict to make sure order is not taken into account.
+        params = urls.url_decode(request.httprequest.query_string)
+        canonical_params = canonical_params or werkzeug.datastructures.MultiDict()
+        # Compare URL has to be done without lang in URL because the lang code
+        # has been removed from request.httprequest.base_url during dispatch.
```

Comment:
bof :/ can't we use e.g. 

current_url = request.rerouting and request.rerouting[0] or request.httprequest.path

Reply:
As discussed, used `root_url` + rerouting stuff to get rid of `lang_in_url`